propolis/src/hw/virtio/viona.rs

```diff
@@ -368,6 +362,104 @@ impl VionaHdl {
     }
 }
 
+// This is an ugly hack to work around tokio's inability to poll for event
+// readiness on states other than POLLIN/POLLOUT, since viona communicates
+// changes to in-kernel ring interrupt state with POLLRDBAND.  In the short
+// term, we can translate that to POLLIN using nested epoll.  The viona fd is
+// added to an epoll handle, subscribing to EPOLLRDBAND.  When that condition is
+// met for the device, epoll will generate an event, making the epoll fd itself
+// readable.  We can subscribe to that using the normal tokio event system.
+//
+// In the long term, viona should probably move to something like eventfd to
+// make polling on those ring interrupt events more accessible.
```

Comment:
What about patching mio?

https://github.com/tokio-rs/mio/blob/master/src/sys/unix/selector/epoll.rs seems to consider "readable" to be "EPOLLIN" or "EPOLLRDHUP" - why wouldn't `EPOLLRDBAND` be part of this too?

Reply:
I'm a bit grumpy that there's no way to actually differentiate them as a consumer of `mio`.  My eventual plan is to change how viona does these notifications.  `POLLRDBAND` is rather awkward.